docs/20_smart-contracts/30_tutorials/10_creating-a-token.md

```diff
@@ -0,0 +1,631 @@
+#include <eosio/asset.hpp>
+#include <eosio/singleton.hpp>
+using namespace eosio;
+
+CONTRACT token : public contract {
+
+    public:
+    using contract::contract;
+
+    // TODO: Add actions
+};
+```
+
+## Creating the actions
+
+Our token contract will have three actions: 
+
+```c++
+    ACTION issue(name to, asset quantity){
+        
+    }
+    
+    ACTION burn(name owner, asset quantity){
+        
+    }
+    
+    ACTION transfer(name from, name to, asset quantity, std::string memo){
+        
+    }
+```
+
+Add them to your contract and then let's dig into each action and see what they do, and what parameters they take.
+
+### Issue
+
+The `issue` action creates new tokens and adds them to an account's balance and to the total supply. 
+
+It takes two parameters:
+- **to**: The account to which the tokens will be issued
+- **quantity**: The amount of tokens to issue
+
+### Burn
+
+The `burn` action removes tokens from an account's balance and the total supply. 
+
+It takes two parameters:
+- **owner**: The account that will be burning the tokens
+- **quantity**: The amount of tokens to burn
+
+### Transfer
+
+The `transfer` action transfers tokens from one account to another. 
+
+It takes four parameters:
+- **from**: The account sending the tokens
+- **to**: The account receiving the tokens
+- **quantity**: The amount of tokens to transfer
+- **memo**: A memo to include with the transfer
+
+## Setting the symbol and precision
+
+Every fungible token has a **symbol** and a **precision**.
+
+The **symbol** is an identifier for the token (like EOS, BTC, or in our case GOLD), and the **precision** is the number of decimal places that the token supports.
+We are going to add a constant variable to our contract to define the `symbol` and `precision` of our token.
+
+Add this above the `issue` action:
+
+```c++
+    const symbol TOKEN_SYMBOL = symbol(symbol_code("GOLD"), 4);
+    
+    ACTION issue ...
+```
+
+The above lines means that we are going to create a token with the symbol `GOLD` and a precision of `4`.
+
+It will look like `100.0000 GOLD` or `0.0001 GOLD`.
+
+## Adding the data structures
+
+Now that you have your actions defined, let's add the data structures that will be used to store the token's data.
+
+Put this below the `TOKEN_SYMBOL` you just added.
+
+```c++
+    const symbol TOKEN_SYMBOL = symbol(symbol_code("GOLD"), 4);
+
+    TABLE balance {
+        name     owner;
+        asset    balance;
+
+        uint64_t primary_key()const { 
+            return owner.value; 
+        }
+    };
+    
+    using balances_table = multi_index<"balances"_n, balance>;
+    
+```
+
+You just created a `balance` structure which defines the data that will be stored in the `balances` table.
+Then, you defined the `balances_table` type which is the definition of a table that will store rows of the `balance` model.
+Later you will construct that table into an object that you can use to store and retrieve data from the blockchain.
```

Comment:
the way it is formulated it sounds off to me and vague which could lead to confusion.
I'd say it like this:

Later you will use the `balances_table` type to instantiate a reference to the `balances` table, and use that reference to store and retrieve data to/from the blockchain.